lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -409,17 +405,21 @@ private JourneyResponse buildF2FNoMatchResponse(
                 auditEventUser);
         var mitigatedCI = ciMitUtilityService.hasMitigatedContraIndicator(contraIndicators);
         if (mitigatedCI.isPresent()) {
-            var mitigationRoute = ciMitUtilityService.getMitigatedCiJourneyStep(mitigatedCI.get());
-            if (mitigationRoute.isPresent()) {
-                JourneyResponse journeyResponse = mitigationRoute.get();
-                if (!JOURNEY_ENHANCED_VERIFICATION_PATH.equals(journeyResponse.getJourney())) {
-                    throw new UnsupportedMitigationRouteException(
-                            String.format(
-                                    "Unsupported mitigation route: %s",
-                                    journeyResponse.getJourney()));
-                }
-                return JOURNEY_ENHANCED_VERIFICATION_F2F_FAIL;
+            var mitigationJourney =
+                    ciMitUtilityService
+                            .getMitigatedCiJourneyStep(mitigatedCI.get())
+                            .map(JourneyResponse::getJourney)
+                            .orElseThrow(
+                                    () ->
+                                            new MitigationRouteException(
+                                                    String.format(
+                                                            "Empty mitigation route for mitigated CI: %s",
+                                                            mitigatedCI.get())));
+            if (!JOURNEY_ENHANCED_VERIFICATION_PATH.equals(mitigationJourney)) {
+                throw new MitigationRouteException(
+                        String.format("Unsupported mitigation route: %s", mitigationJourney));
             }
```

Comment:
(I hate the special-casing here, but fine to leave...)

Reply:
We could take this check out - it's only really here to catch a situation which should never really happen. Perhaps I was being over cautious adding it in the first place.

Reply:
No I think it makes sense - we need some kind of capability to say 'run this step before emitting the same event again'. A problem for another day...